doc/guides/V8-api-for-node.md

```diff
@@ -0,0 +1,210 @@
+## V8 API usage for Node.js
+
+v8 docs, not particularly useful:
+- https://v8.dev/docs
+
+Some API info at https://v8.dev/docs/embed
+
+- An isolate is a VM instance with its own heap.
+  Node has one isolate.
+  Can get from `Local<Object>->GetIsolate()`, `context->GetIsolate()`
+
+- `Local<...>`: A local handle is a pointer to an object. All V8 objects are
+  accessed using handles, they are necessary because of the way the V8 garbage
+  collector works.  Local handles can only be allocated on the stack, in a
+  scope, not with new, and will be deleted as stack unwinds, usually with a
+  scope.
+  - XXX don't see us create scopes much in node, why?
+  - XXX locals go into a HandleScope, for auto-dispose? Always, or by default?
+- Persistent handles last past C++ functions.
+  - PersistentBase::SetWeak trigger a callback from the garbage collector when
+    the only references to an object are from weak persistent handles.
+  - A UniquePersistent<SomeType> handle relies on C++ constructors and
+    destructors to manage the lifetime of the underlying object.
+  - A Persistent<SomeType> can be constructed with its constructor, but must be
+    explicitly cleared with Persistent::Reset.
+- Eternal is a persistent handle for JavaScript objects that are expected to
+  never be deleted. It is cheaper to use because it relieves the garbage
+  collector from determining the liveness of that object.
+- A handle scope can be thought of as a container for any number of handles.
+  When you've finished with your handles, instead of deleting each one
+  individually you can simply delete their scope.
+- EscapableHandleScope: Locals can't be returned, their scope will delete them,
+  so you need an escable scope, and to `return scope.Escape(...the local)`. It
+  scopes the locals into the enclosing scope and returns a local for that scope.
+
+Local, MaybeLocal, Maybe, Value, oh my...
+
+https://groups.google.com/forum/#!topic/v8-users/gQVpp1HmbqM
+
+- MaybeLocal may be "empty", basically not contain a pointer of its type.  See
+  `class MaybeLocal`, has some useful notes on why, but basically its returned
+  by anything that might fail to have a value.
+- If you know that the MaybeLocal has a value, then call ToLocalChecked() and
+  node will abort with CHECK() if you are wrong.
+- Otherwise, you have to call `bool ToLocal(Local<S>* out)`, and check the
+  return value to see if there was a value. Or call IsEmpty() to check.
+
+- Maybe is similar, but doesn't hold a Local, just a value of T. "Just" means it
+  "just has a value", a bizarrely named Haskellism :-(. It has a To() and
+  ToChecked() similar to MaybeLocal.
+- A common node idiom is to make a seemingly side-effect free call to
+  `.FromJust()` after `->Set()`, which will crash node if the Set failed.
```

Comment:
We do use FromJust() a lot but V8 has Check() these days which is arguably more appropriate: returns void and the name makes it clearer that it can fail. Maybe we should switch over in one fell swoop...

Reply:
https://github.com/nodejs/node/pull/27162